diff -uw external/FATFileSystem/FATFileSystem.h external.new/FATFileSystem/FATFileSystem.h
--- external/FATFileSystem/FATFileSystem.h	2011-09-18 01:05:29.000000000 -0700
+++ external.new/FATFileSystem/FATFileSystem.h	2011-09-12 12:36:08.000000000 -0700
@@ -41,7 +41,7 @@
 	virtual int remove(const char *filename);
 	virtual int format();
         virtual DirHandle *opendir(const char *name);
-        virtual int mkdir(const char *name, mode_t mode);
+    virtual int mkdir(const char *name, int mode);
 	
     FATFS _fs;            					// Work area (file system object) for logical drive	
     static FATFileSystem *_ffs[_DRIVES];	// FATFileSystem objects, as parallel to FatFs drives array
Common subdirectories: external/FATFileSystem/LPC1768 and external.new/FATFileSystem/LPC1768
Common subdirectories: external/FATFileSystem/LPC2368 and external.new/FATFileSystem/LPC2368
