diff -uw external/mbed/DirHandle.h external.new/mbed/DirHandle.h
--- external/mbed/DirHandle.h	2011-09-18 01:05:23.000000000 -0700
+++ external.new/mbed/DirHandle.h	2011-09-12 12:36:08.000000000 -0700
@@ -76,7 +76,7 @@
      *  location - The location to seek to. Must be a value returned
      *   by telldir.
      */
-    virtual void seekdir(off_t location) { }
+    virtual void seekdir(off_t location) { (void)location; }
 
 };
 
diff -uw external/mbed/FileLike.h external.new/mbed/FileLike.h
--- external/mbed/FileLike.h	2011-09-18 01:05:23.000000000 -0700
+++ external.new/mbed/FileLike.h	2011-09-12 12:36:08.000000000 -0700
@@ -24,7 +24,7 @@
      * Variables
      *  name - The name to use to open the file.
      */
-    FileLike(const char *name) : Base(name) { }
+    FileLike(const char *pName) : Base(pName) { }
     virtual ~FileLike();
 
 };
diff -uw external/mbed/FileSystemLike.h external.new/mbed/FileSystemLike.h
--- external/mbed/FileSystemLike.h	2011-09-18 01:05:23.000000000 -0700
+++ external.new/mbed/FileSystemLike.h	2011-09-12 12:36:08.000000000 -0700
@@ -39,7 +39,7 @@
      * Variables
      *  name - The name to use for the filesystem.
      */
-    FileSystemLike(const char *name) : Base(name) {}
+    FileSystemLike(const char *pName) : Base(pName) {}
 
     /* Function open
      *
@@ -59,7 +59,7 @@
      *  filename - the name of the file to remove.
      *  returns - 0 on success, -1 on failure.
      */
-    virtual int remove(const char *filename) { return -1; };
+    virtual int remove(const char *filename) { (void)filename; return -1; };
 
     /* Function rename
      *  Rename a file in the filesystem.
@@ -69,7 +69,7 @@
      *  newname - the name to rename it to.
      *  returns - 0 on success, -1 on failure.
      */
-    virtual int rename(const char *oldname, const char *newname) { return -1; };
+    virtual int rename(const char *oldname, const char *newname) { (void)oldname; (void)newname; return -1; };
 
     /* Function opendir
      *  Opens a directory in the filesystem and returns a DirHandle
@@ -80,7 +80,7 @@
      *  returns - A DirHandle representing the directory stream, or
      *   NULL on failure.
      */
-    virtual DirHandle *opendir(const char *name) { return NULL; };
+    virtual DirHandle *opendir(const char *pName) { (void)pName; return NULL; };
 
     /* Function mkdir
      *  Creates a directory in the filesystem.
@@ -90,7 +90,7 @@
      *  mode - The permissions to create the directory with.
      *  returns - 0 on success, -1 on failure.
      */
-    virtual int mkdir(const char *name, mode_t mode) { return -1; }
+    virtual int mkdir(const char *pName, int mode) { (void)pName; (void)mode; return -1; }
 
     // TODO other filesystem functions (mkdir, rm, rn, ls etc)
     
Common subdirectories: external/mbed/LPC1768 and external.new/mbed/LPC1768
Common subdirectories: external/mbed/LPC2368 and external.new/mbed/LPC2368
diff -uw external/mbed/PeripheralNames.h external.new/mbed/PeripheralNames.h
--- external/mbed/PeripheralNames.h	2011-09-18 01:05:23.000000000 -0700
+++ external.new/mbed/PeripheralNames.h	2011-09-12 12:36:08.000000000 -0700
@@ -15,15 +15,14 @@
 extern "C" {
 #endif 
 
-typedef enum UARTName UARTName;
 enum UARTName {
     UART_0 = (int)LPC_UART0_BASE
     , UART_1 = (int)LPC_UART1_BASE
     , UART_2 = (int)LPC_UART2_BASE
     , UART_3 = (int)LPC_UART3_BASE
 };
+typedef enum UARTName UARTName;
 
-typedef enum ADCName ADCName;
 enum ADCName {
     ADC0_0 = 0
     , ADC0_1
@@ -34,26 +33,26 @@
     , ADC0_6
     , ADC0_7
 };
+typedef enum ADCName ADCName;
 
-typedef enum DACName DACName;
 enum DACName {
     DAC_0 = 0
 };
+typedef enum DACName DACName;
 
-typedef enum SPIName SPIName;
 enum SPIName {
     SPI_0 = (int)LPC_SSP0_BASE
     , SPI_1 = (int)LPC_SSP1_BASE
 };
+typedef enum SPIName SPIName;
 
-typedef enum I2CName I2CName;
 enum I2CName {
     I2C_0 = (int)LPC_I2C0_BASE
     , I2C_1 = (int)LPC_I2C1_BASE
     , I2C_2 = (int)LPC_I2C2_BASE
 };
+typedef enum I2CName I2CName;
 
-typedef enum PWMName PWMName;
 enum PWMName {
     PWM_1 = 1
     , PWM_2 
@@ -62,20 +61,21 @@
     , PWM_5 
     , PWM_6 
 };
+typedef enum PWMName PWMName;
 
-typedef enum TimerName TimerName;
 enum TimerName {
     TIMER_0 = (int)LPC_TIM0_BASE
     , TIMER_1 = (int)LPC_TIM1_BASE
     , TIMER_2 = (int)LPC_TIM2_BASE
     , TIMER_3 = (int)LPC_TIM3_BASE
 };
+typedef enum TimerName TimerName;
 
-typedef enum CANName CANName;
 enum CANName { 
      CAN_1 = (int)LPC_CAN1_BASE,
      CAN_2 = (int)LPC_CAN2_BASE 
 }; 
+typedef enum CANName CANName;
 
 #define STDIO_UART_TX     USBTX
 #define STDIO_UART_RX     USBRX
diff -uw external/mbed/PinNames.h external.new/mbed/PinNames.h
--- external/mbed/PinNames.h	2011-09-18 01:05:23.000000000 -0700
+++ external.new/mbed/PinNames.h	2011-09-12 12:36:08.000000000 -0700
@@ -15,7 +15,6 @@
 extern "C" {
 #endif 
 
-typedef enum PinName PinName;
 enum PinName {
 
     // LPC Pin Names
@@ -80,14 +79,15 @@
     , NC = (int)0xFFFFFFFF
 
 };
+typedef enum PinName PinName;
 
-typedef enum PinMode PinMode;
 enum PinMode {
     PullUp = 0
     , PullDown = 3
     , PullNone = 2
     , OpenDrain = 4
 };
+typedef enum PinMode PinMode;
 
 #ifdef __cplusplus
 }
diff -uw external/mbed/PortNames.h external.new/mbed/PortNames.h
--- external/mbed/PortNames.h	2011-09-18 01:05:23.000000000 -0700
+++ external.new/mbed/PortNames.h	2011-09-12 12:36:08.000000000 -0700
@@ -10,7 +10,6 @@
 extern "C" {
 #endif
 
-typedef enum PortName PortName;
 enum PortName {
     Port0 = 0
     , Port1 = 1
@@ -18,6 +17,7 @@
     , Port3 = 3
     , Port4 = 4
 };
+typedef enum PortName PortName;
 
 #ifdef __cplusplus
 }
diff -uw external/mbed/can_helper.h external.new/mbed/can_helper.h
--- external/mbed/can_helper.h	2011-09-18 01:05:23.000000000 -0700
+++ external.new/mbed/can_helper.h	2011-09-12 12:36:08.000000000 -0700
@@ -10,19 +10,18 @@
 extern "C" {
 #endif
 
-typedef enum CANFormat CANFormat;
 enum CANFormat {
     CANStandard = 0,
     CANExtended = 1
 };
+typedef enum CANFormat CANFormat;
 
-typedef enum CANType CANType;
 enum CANType {
     CANData   = 0,
     CANRemote = 1
 };
+typedef enum CANType CANType;
 
-typedef struct CAN_Message CAN_Message;
 struct CAN_Message {
     unsigned int   id;                 // 29 bit identifier
     unsigned char  data[8];            // Data field
@@ -30,6 +29,7 @@
     CANFormat      format;             // 0 - STANDARD, 1- EXTENDED IDENTIFIER
     CANType        type;               // 0 - DATA FRAME, 1 - REMOTE FRAME
 };
+typedef struct CAN_Message CAN_Message;
 
 #ifdef __cplusplus
 };
